extensions/amp-viewer-integration/0.1/messaging/messaging.js

```diff
@@ -392,3 +391,39 @@ export class Messaging {
     return err ? (err.message ? err.message : String(err)) : 'unknown error';
   }
 }
+
+/**
+ * Performs a handshake and initializes messaging.
+ * @param {!Window} source
+ * @param {!HTMLIFrameElement} iframe
+ * @param {string} origin
+ * @return {!Promise<!Messaging>}
+ */
+export function initMessaging(source, iframe, origin) {
```

Comment:
Let's immediately refactor the origin of this code to use this function, so that we can take advantage of the related test base right away.

Reply:
Sure, will send another PR for that since it's from a different repo.

Reply:
I meant from the this repo. In response to your description:

> Adds initMessaging helper function (most of this logic is from here).

I assume we want to refactor the place "of this logic"?

Reply:
Yes, but the function comes from / is used by github.com/ampproject/amp-viewer. I'll update that repo when we merge this and publish the new version to npm.

Reply:
Ok. Got it. But can you clarify who is supposed to call this function? Assuming two parties involved (viewer and doc) either could, in theory, try to initialize messaging and all that. So we need to be very clear about direction of the handshake. Also, I'm not seeing any "initiation" actions in this method. So the old name "waitFor..." expresses what it does better.

Reply:
Added polling method too and changed names.